Refactor and also send emails on processing and queued

pathakh opened the pull request:
I still have to write some partial specs for the mailers.  And there are existing specs that continue to work after deleting calls to send emails.

hoenth commented on app/mailers/nfg_csv_importer/import_mailer.rb:
```diff
@@ -4,15 +4,15 @@ class NfgCsvImporter::ImportMailer < ActionMailer::Base
     mail(
       to: @recipient.email,
-      subject: "Your #{@import.import_type} import is #{status}!",
+      subject: "Your #{@import.import_type} import is #{@import.status}#{show_excitement}",
```
We should consider whether "import_type" is what we want to show here. Many org users may not know that they are importing an individual_donations import. I think we should take it out for now and just say "Your import is #{@import.status}#"

hoenth commented on app/mailers/nfg_csv_importer/import_mailer.rb:
```diff
@@ -45,7 +33,12 @@ def imported_for(import)
+  end
+
   def url_options(imported_for)
+    return {} if @import.deleted?
```
Why are url_options an empty hash if the import is deleted?

pathakh replied:
url_options wasn't present in the method that was deleted.  The below method was deleted, the only difference between this and the one used for everything else was the url options.

```
  def send_destroy_result(import)	  def path
    @import = import	    @path ||= "#{self.controller_path}/#{self.action_name}_#{@import.status}"
    @recipient = import.imported_by	
    @imported_for = imported_for(@import)	

     mail(	
      to: @recipient.email,	
      subject: "Your #{@import.import_type} import has been deleted.",	
      from: NfgCsvImporter.configuration.from_address,	
      reply_to: NfgCsvImporter.configuration.reply_to_address,	
      skip_premailer: true	
    ) do |format|	
      format.html { inlined_html }	
    end
```

hoenth replied:
@pathakh OK, then I don't think we need to set url options to an empty hash then. It just won't be used in the delete message. By setting it to an empty hash, it implies that it can't have values.

pathakh replied:
ah i see what you mean, I just saw that url options gets used in creating a link, and that link is not present in destroy.  I'll remove this line.

hoenth commented on app/views/nfg_csv_importer/import_mailer/send_import_result/_steps.html.haml:
```diff
@@ -0,0 +1,12 @@
+- import_mailer_presenter = NfgCsvImporter::Mailers::ImportMailerPresenter.new(@import, self)
```
Wouldn't we pass this presenter in as a local assignment?

hoenth commented on app/views/nfg_csv_importer/import_mailer/send_import_result_complete.html.haml:
```diff
@@ -3,9 +3,8 @@
 - locales_scope = [:mailers, :nfg_csv_importer, :send_import_result_mailer]
```
What was this file renamed to? It didn't display the new full name in the PR

hoenth replied:
If this file is only now for completed imports then we don't need the .show_successful_import_illustration as it would really only be checking if the file had errors, so we could just check for that specifically

hoenth commented on app/views/nfg_csv_importer/import_mailer/send_import_result_processing.html.haml:
```diff
@@ -0,0 +1,21 @@
+- import_mailer_presenter = NfgCsvImporter::Mailers::ImportMailerPresenter.new(@import, self)
```
Seems like we should set this presenter in the mailer instead of in the view.

hoenth commented on app/views/nfg_csv_importer/import_mailer/send_import_result_processing.html.haml:
```diff
@@ -0,0 +1,21 @@
+- import_mailer_presenter = NfgCsvImporter::Mailers::ImportMailerPresenter.new(@import, self)
+
+- locales_scope = [:mailers, :nfg_csv_importer, :send_import_result_mailer]
```
Same here, set this in the mailer, since it is the same for all of them

pathakh commented:
@hoenth I've addressed the comments.  

Also, i think we need separate stories to add specs as most of the files are not tested in the entire project.

hoenth commented:
@pathakh we just want to add specs for the mailers individually, to capture if there are any missing values and that they are using the import status correctly. We also want to verify that the emails get sent at the appropriate times. The tricky one will be the 'queued' email, as there are a couple places where an import can be queued, and we don't want to send any of these for the send to nfg emails. 

As to other specs, there are currently 416 tests in this project, so I am not clear what you mean by "files are not tested in the entire project"

pathakh commented:
@hoenth all the specs are done.  I'm not sure if you would like to look at it again.  If not, i can go ahead and merge it.

pathakh commented:
@hoenth i've written the specs for the controllers and jobs that call import mailer.  And also the view specs.  I'll go ahead and check it into the main branch since i feel this is complete. 
Also, you mentioned there is already a check against sending an email for "send_to_nfg" email.  I couldn't find that code though however.

hoenth commented:
@pathakh The importing gem shouldn't know anything about send_to_nfg, as that is completely configured in DM, so there shouldn't be specs here that test that. It is just a different file_origination_type whose post upload of pre_processing files job sends the email.

pathakh commented:
> @pathakh The importing gem shouldn't know anything about send_to_nfg, as that is completely configured in DM, so there shouldn't be specs here that test that. It is just a different file_origination_type whose post upload of pre_processing files job sends the email.

sounds good thanks @hoenth
